include/fs/fs.h

```diff
@@ -170,20 +194,35 @@ struct fs_file_system_t {
 #define FS_SEEK_END	2	/* Seek from end of file.  */
 #endif
 
-
 /**
- * @brief File open
+ * @brief File open extended
  *
- * Opens an existing file or create a new one and associates
- * a stream with it.
+ * Opens or crates, if does not exist file, depending on mode provided
+ * and associates a stream with it.
  *
  * @param zfp Pointer to file object
  * @param file_name The name of file to open
+ * @param flags The mode flags, if not provided default to FO_CREATE | FO_RDWR
+ *
+ * @p flags can be empty, or combination of one or more of following flags:
+ *   FO_READ		open for read
+ *   FO_WRITE		open for write
+ *   FO_RDWR		open for read/write (<tt>FO_READ | FO_WRITE</tt>)
+ *   FO_CREATE		crete file if it does not exist
+ *   FO_APPEND		open file at the end
+ *
+ * Flag FO_CREATE and FO_APPEND require FO_WRITE, otherwise EINVAL will be
+ * returned.
  *
  * @retval 0 Success
- * @retval -ERRNO errno code if error
+ * @retval -EINVAL when bad combination of flags is given
+ *	   -EINVAL when bad file name is given
+ *	   -NOENT  when file path is not possible (bad mount point)
+ *	   -ERRNO  other errno code, deending of file system backend
  */
-int fs_open(struct fs_file_t *zfp, const char *file_name);
+int fs_open(struct fs_file_t *zfp, const char *file_name, int flags);
```

Comment:
I don't particularly like seeing signed types used to hold bitfields.  I guess existing practice in legacy APIs like POSIX might justify it.

Reply:
Done